cdap/resource_gcs_artifact.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdap
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"regexp"
+
+	"cloud.google.com/go/storage"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var bucketPathRE = regexp.MustCompile(`^gs://(.+)/(.+)$`)
+
+func resourceGCSArtifact() *schema.Resource {
```

Comment:
Add a note why we need this and not directly use the TF's GCS object resource, and maybe in the future to migrate to use that one directly.

Reply:
Done.